src/modules/commons/components/searchableSelect/SearchableSelect.js

```diff
@@ -0,0 +1,303 @@
+			evt.target.classList.remove('hovered');
+		};
+
+		const onOptionChosen = (evt) => {
+			// Prevents global click handler to trigger s. onInitialize()
+			evt.stopPropagation();
+			this.#hoverOption(evt.target);
+			this.#confirmAction();
+		};
+
+		return html`
+			<style>
+				${css}
+			</style>
+
+			<div class="searchable-select" @pointerenter=${onPointerEnter} @pointerleave=${onPointerLeave} @click=${onSearchInputClicked}>
+				<div class="search-input-container">
+					<input id="search-input" type="text" .placeholder=${placeholder} .value=${search} @input=${onSearchInputChange} />
+					<div class="caret-fill-down"></div>
+				</div>
+				<div class="dropdown hidden">
+					${availableOptions.slice(0, maxEntries).map(
+						(item, index) =>
+							html`<div
+								class="option"
+								@click=${onOptionChosen}
+								@pointerenter=${onPointerEnterOption}
+								@pointerleave=${onPointerLeaveOption}
+								.value=${index}
+							>
+								<span>${item}</span>
+							</div>`
+					)}
+				</div>
+			</div>
+		`;
+	}
+
+	#hoverOption(target) {
+		// Due to keyboard events, multiple options can be hovered. The following line prevents it.
+		this.shadowRoot.querySelectorAll('.option.hovered').forEach((option) => option.classList.remove('hovered'));
+		target.classList.add('hovered');
+	}
+
+	#updateOptionsFiltering(model) {
+		const { search, options } = model;
+		const ucSearchTerm = search.toUpperCase();
+		const filteredOptions = [];
+
+		for (const option of options) {
+			if (option.toUpperCase().indexOf(ucSearchTerm) > -1) {
+				filteredOptions.push(option);
+			}
+		}
+
+		return { ...model, availableOptions: filteredOptions };
+	}
+
+	#hideDropdown() {
+		const itemsContainer = this.shadowRoot.querySelector('.dropdown');
+		itemsContainer.classList.remove('visible');
+		itemsContainer.classList.add('hidden');
+
+		this.#keyActionMapper.deactivate();
+	}
+
+	#showDropdown() {
+		const itemsContainer = this.shadowRoot.querySelector('.dropdown');
+		itemsContainer.classList.add('visible');
+		itemsContainer.classList.remove('hidden');
+
+		// Duplicate event safety
+		this.#keyActionMapper.deactivate();
+		this.#keyActionMapper
+			.addForKeyUp('Escape', () => this.#cancelAction())
+			.addForKeyUp('Enter', () => this.#confirmAction())
+			.addForKeyUp('ArrowUp', () => this._chooseNextOption(true))
+			.addForKeyUp('ArrowDown', () => this._chooseNextOption());
+
+		this.#keyActionMapper.activate();
+	}
+
+	#cancelAction() {
+		this.#hideDropdown();
+
+		if (!this.selected) {
+			this.signal(Update_Search, '');
+		}
+	}
+
+	#confirmAction() {
+		const hoveredOption = this.shadowRoot.querySelector('.option.hovered');
+		if (!hoveredOption) return;
+
+		this.#hideDropdown();
+		// @ts-ignore
+		this.signal(Update_Selected, hoveredOption.value);
+	}
+
+	#dispatchChangeEvents() {
+		const { selected, availableOptions } = this.getModel();
+		const eventData = {
+			availableOptions: availableOptions,
```

Comment:
`availableOptions` contains only one entry and not as expected all possible entries